iron-fit-behavior.js

```diff
@@ -12,6 +12,70 @@ import '@polymer/polymer/polymer-legacy.js';
 
 import {dom} from '@polymer/polymer/lib/legacy/polymer.dom.js';
 
+// IE11 has a bug where an element with (1) `overflow: auto;`, (2) size based on
+// its content's natural size, (3) absolute positioning (either `absolute` or
+// `fixed`), and (4) use `max-width` to constrain its width will place its
+// vertical scrollbar outside the area constrained by `max-width`.
+const hasVerticalScrollbarMaxWidthBug = (() => {
+  let hasVerticalScrollbarMaxWidthBug = undefined;
+  return () => {
+    if (hasVerticalScrollbarMaxWidthBug === undefined) {
+      const container = document.createElement('div');
+      Object.assign(container.style, {
+        overflow: 'auto',
+        position: 'fixed',
+        left: '0px',
+        top: '0px',
+        maxWidth: '100px',
+        maxHeight: '100px',
+      });
+
+      const content = document.createElement('div');
+      content.style.width = '200px';
+      content.style.height = '200px';
+      container.appendChild(content);
+
+      document.body.appendChild(container);
+      hasVerticalScrollbarMaxWidthBug =
+          Math.abs(container.offsetWidth - 100) > 1;
+      document.body.removeChild(container);
+    }
+
+    return hasVerticalScrollbarMaxWidthBug;
+  };
+})();
+
+const scrollbarSize = (() => {
```

Comment:
This probably wouldn't work in all cases since css can change scrollbar width and the scrollbar can be different on the body compared to the dropdown

Reply:
Yeah, that's true. Fortunately, this is only used when the browser has the vertical scrollbar + `max-width` bug mentioned above and IE 11 doesn't support modifying the scrollbar size (AFAICT), so I think it's ok to cache this value. However, I combined `scrollbarSize` and `hasVerticalScrollbarMaxWidthBug` so that nobody later might come along and try to use `scrollbarSize` in some other case where they might really need to read it live.